cmd/kops/delete_cluster.go

```diff
@@ -133,7 +133,7 @@ func RunDeleteCluster(ctx context.Context, f *util.Factory, out io.Writer, optio
 			}
 		}
 
-		allResources, err := resourceops.ListResources(cloud, clusterName, options.Region)
+		allResources, err := resourceops.ListResources(cloud, cluster, options.Region)
```

Comment:
I believe that in some scenarios when deleting a cluster we may not have access to the cluster spec, for example if the `--external` flag is set here in which case Kops only searches for resources to delete based on the `KubernetesCluster` tag value.

I see that `pkg/resources/azure` only uses `.Spec.CloudConfig.Azure.ResourceGroupName`

I'm wondering how we could best handle this situation. We could start off with not allowing `--external` on `kops delete cluster` for azure clusters and try to add support for this later.

Reply:
My mistake, I misread the code right above this. It turns out `--external` only works with AWS (in fact it assumes the cluster is on AWS -- we should definitely fix that but that won't hold up this PR)

In general the AWS cloud provider has two tag formats that it uses: `KubernetesCluster: <clustername>` and `kubernetes.io/cluster/<clustername>: <owned|shared>`. For resources that can be shared between multiple clusters like networks, the network can be tagged with the second format for each cluster using the network, allowing things like `delete cluster` to ensure it doesn't delete resources shared with other clusters.

```
kubernetes.io/cluster/cluster1.k8s.local: shared
kubernetes.io/cluster/cluster2.k8s.local: shared
```

Reply:
Got it! Thanks for the detailed explanation!